src/components/mdredactor/MdRedactor.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState, useEffect } from "react";
+import MdEditor, { Plugins } from "react-markdown-editor-lite";
+import MarkdownIt from "markdown-it";
+import "react-markdown-editor-lite/lib/index.css";
+import "./splittstyle.css";
+import { DeliveredProcedureOutlined } from "@ant-design/icons";
+
+const mdParser = new MarkdownIt({
+  html: true,
+  linkify: true,
+  typographer: true,
+});
+
+MdEditor.addLocale("de-DE", {
+  btnHeader: "Überschrift",
+  btnClear: "Löschen",
+  btnBold: "Fett",
+  btnItalic: "Kursiv",
+  btnUndo: "Rückgängig machen",
+  btnRedo: "Wiederholen",
+  btnUnderline: "Unterstrichen",
+  btnStrikethrough: "Durchgestrichen",
+  btnUnordered: "Aufzählungsliste",
+  btnOrdered: "Nummerierte Liste",
+  btnQuote: "Zitat",
+  btnLink: "Link",
+});
+MdEditor.useLocale("de-DE");
+
+const ViewMode = (props) => {
+  const [isWriteActive, setIsWriteActive] = useState(true);
+  const [isPreviewActive, setIsPreviewActive] = useState(false);
+  const handleClickPreview = () => {
+    props.editor.setView({
+      md: false,
+      menu: true,
+      html: true,
+    });
+    setIsWriteActive(false);
+    setIsPreviewActive(true);
+  };
+  const handleClickWrite = () => {
+    props.editor.setView({
+      md: true,
+      menu: true,
+      html: false,
+    });
+    setIsWriteActive(true);
+    setIsPreviewActive(false);
+  };
+
+  return (
+    <div>
+      <div
+        className={isWriteActive ? "mode-btn-toogle-active" : "mode-btn-toogle"}
+        onClick={handleClickWrite}
+        // title="Bearbeitenmodus"
+      >
+        Bearbeiten
+      </div>
+      <div
+        className={
+          isPreviewActive ? "mode-btn-toogle-active" : "mode-btn-toogle"
+        }
+        onClick={handleClickPreview}
+        // title="Vorschaumodus"
+      >
+        Vorschau
+      </div>
+    </div>
+  );
+};
+
+ViewMode.defaultConfig = {};
+ViewMode.align = "right";
+ViewMode.pluginName = "viewmode";
+
+MdEditor.use(ViewMode, {});
+
+const SaveDocumentBtn = (props) => {
+  const handleSaveDocument = () => {
+    const mdDoc = props.editor.getMdValue();
+    console.log("xxx mdDoc", { mdDoc });
+    props.config.getDocFn();
+  };
+
+  return (
+    <div>
+      <span
+      // className="button"
+      >
+        <DeliveredProcedureOutlined
+          className="save-button"
+          title="Dokument speichern"
+          onClick={handleSaveDocument}
+          style={{ marginTop: "6px", marginLeft: "5px", cursor: "pointer" }}
+        />
+      </span>
+    </div>
```

Comment:
we probably don't need that button but if we would don't put the onClick on the icon and remove the span. either use <button></button> with the onClick or use <div role="button" onClick={}></div> and move the classname to the button itself and not the icon. You also use style + className for the icon. Stick to one or the other but not both if you don't need to